examples/different_integrators/smoothed_pss.m

```diff
@@ -0,0 +1,121 @@
+% BSD 2-Clause License
+
+% Copyright (c) 2022, Armin Nurkanović, Jonathan Frey, Anton Pozharskiy, Moritz Diehl
+
+% Redistribution and use in source and binary forms, with or without
+% modification, are permitted provided that the following conditions are met:
+
+% 1. Redistributions of source code must retain the above copyright notice, this
+%    list of conditions and the following disclaimer.
+
+% 2. Redistributions in binary form must reproduce the above copyright notice,
+%    this list of conditions and the following disclaimer in the documentation
+%    and/or other materials provided with the distribution.
+
+% THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+% AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+% IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+% DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+% FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+% DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+% SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+% CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+% OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+% OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+% This file is part of NOSNOC.
+
+%
+%%
+clear all
+clc
+close all
+import casadi.*
+%%
+plot_integrator_output = 1;
+plot_continious_time_sol = 1;
+%% discretization settings
+T_sim = pi/2;
+N_sim  = 29;
+N_finite_elements = 2;
+R_osc  = 1;
+
+%% Init
+problem_options = nosnoc.Options();
+solver_options = nosnoc.solver.Options();
+model = nosnoc.model.Pss();
+%% settings
+
+% Select matlab ode solver
+solver_options.matlab_ode_solver = 'ode23s';
```

Comment:
should be default (i would even add a warning if ode15s or oder23s are not used, as they perform very bad if e.g. there is a sliding mode as system is everywhere stiff.
also a refernece for the new class can be chapter 4 of my phd thesis. there are the pros and cons of this simple approach discussed